AssertionError: daemonic processes are not allowed to have children

Agent cannot gather container stat and print traceback like below. It doesn’t happen to every machine

```
ERROR ai.backend.agent.stats [2241513] collect_container_stat(): gather_container_measures() error
Traceback (most recent call last):
  File "python3.13/site-packages/aiomonitor/monitor.py", line 509, in _coro_wrapper
    return await coro
           ^^^^^^^^^^
  File "python3.13/site-packages/ai/backend/agent/docker/intrinsic.py", line 736, in gather_container_measures
    results = await asyncio.gather(*tasks)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "python3.13/site-packages/aiomonitor/monitor.py", line 509, in _coro_wrapper
    return await coro
           ^^^^^^^^^^
  File "python3.13/site-packages/ai/backend/agent/docker/intrinsic.py", line 666, in sysfs_impl
    nstat = await netstat_ns(sandbox_key)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "python3.13/site-packages/ai/backend/agent/docker/intrinsic.py", line 95, in netstat_ns
    result = await loop.run_in_executor(executor, netstat_ns_work, ns_path)
                   ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "uvloop/loop.pyx", line 2747, in uvloop.loop.Loop.run_in_executor
  File "python3.13/concurrent/futures/process.py", line 804, in submit
    self._start_executor_manager_thread()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "python3.13/concurrent/futures/process.py", line 743, in _start_executor_manager_thread
    self._launch_processes()
    ~~~~~~~~~~~~~~~~~~~~~~^^
  File "python3.13/concurrent/futures/process.py", line 770, in _launch_processes
    self._spawn_process()
    ~~~~~~~~~~~~~~~~~~~^^
  File "python3.13/concurrent/futures/process.py", line 780, in _spawn_process
    p.start()
    ~~~~~~~^^
  File "python3.13/multiprocessing/process.py", line 118, in start
    assert not _current_process._config.get('daemon'), \
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: daemonic processes are not allowed to have children
```

--- a/changes/4929.fix.md
+++ b/changes/4929.fix.md
@@ -1,0 +1,1 @@
+Check if Agent is daemon process before query docker netstat

--- a/src/ai/backend/agent/docker/intrinsic.py
+++ b/src/ai/backend/agent/docker/intrinsic.py
@@ -1,5 +1,6 @@
 import asyncio
 import logging
+import multiprocessing
 import os
 import platform
 from collections.abc import Collection, Iterable, Mapping, Sequence
@@ -91,6 +92,19 @@ async def netstat_ns(ns_path: Path):
     # Unfortunately, CPython drops GIL while running IO and does not
     # provide any similar functionality. Therefore we execute namespace
     # dependent operation in the new process.
+
+    # Check if we're already in a daemon process
+    current_process = multiprocessing.current_process()
+    try:
+        is_daemon = current_process.daemon
+    except AttributeError:
+        is_daemon = False
+
+    if is_daemon:
+        # We're in a daemon process, run directly in thread pool
+        # This is less safe but works as a fallback
+        result = await loop.run_in_executor(None, netstat_ns_work, ns_path)
+        return result
     with ProcessPoolExecutor(max_workers=1) as executor:
         result = await loop.run_in_executor(executor, netstat_ns_work, ns_path)
     return result

